dso/model.py

```diff
@@ -133,26 +133,23 @@ class ComponentArtefactId:
     artefact_kind: ArtefactKind | None = None
     references: list[typing.Self] = dataclasses.field(default_factory=list)
 
-    def as_frozenset(self) -> frozenset[str]:
-        props = (
-            self.component_name,
-            self.component_version,
-            self.artefact_kind,
-            frozenset(self.references),
-        )
-
-        if self.artefact:
-            props += (self.artefact.as_frozenset(),)
+    @property
+    def key(self) -> str:
+        artefact_key = self.artefact.key if self.artefact else None
```

Comment:
Currently, `None`-case is just parsed to str so it ends up as `'None'` vs empty-str values `''`. So, in a sense, our meta character for `None`-case is `'None'` right now. I'll move this into a function and create a respective parameter for that case so that this will become explicit rather than implicit.